### PR TITLE
Restructuring of network ACLS

### PR DESCRIPTION
**Pull request by AAYUSH2091**

### CODE REVIEW

**Comment by AAYUSH2091** [comment]
Hi @rohitthakur2590 ,
Could you please review my changes when you get a chance?
I was a bit stuck with the documentation and tried to follow the examples from network.ospf and network.bgp. Let me know if there are areas I should improve or approach differently.

File: tests/integration/targets/acls_remediate/tasks/main.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      ansible.builtin.include_tasks: includes/assertions.yaml
+      loop: "{{ test_resources }}"
+      loop_control:
+        loop_var: filename
+
+    - name: Run network.acls validated content with remediate operation
+      ansible.builtin.include_role:
+        name: network.acls.remediate
+      vars:
+        operations:
```
Comment (rohitthakur2590) [inline tests/integration/targets/acls_remediate/tasks/main.yaml]: I believe we don;t need this anymore

**Review by rohitthakur2590 — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -60,6 +60,8 @@ This platform-agnostic role enables the user to perform acls health checks. User
        `unassigned_acls`
 - This role enables users to create a runtime brownfield inventory with all the acls configurations regarding host vars. These host vars are ansible facts gathered through the *_acls and *_acl_interfaces network resource modules. The tasks offered by this role can be observed below:
 
+
```
Comment (rohitthakur2590) [inline README.md]: is an extra line required here?

File: roles/detect/README.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+- Detect configuration drifts between running configurations and host variables.
+- Support for local and remote data stores.
+- Provide detailed reports of detected changes.
+
+## Variables
+
+| Variable Name        | Default Value | Required | Type | Description                                                   |         Example          |
+|:---------------------|:-------------:|:--------:|:----:|:------------------------------------------------------------- |:------------------------:|
+| `ansible_network_os` | `""`          | no       | str  | Network OS to be used during detect.                          |    `"cisco.ios.ios"`     |
+| `data_store`         | `""`          | yes      | dict | DSpecifies the data store to be used (local or SCM).          | See usage example below. |
```
Comment (rohitthakur2590) [inline roles/detect/README.md]:
```suggestion
| `data_store`         | `""`          | yes      | dict | Specifies the data store to be used (local or SCM).          | See usage example below. |
```

File: roles/gather/README.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  gather_facts: true
+  tasks:
+    - name: Invoke gather role
+      ansible.builtin.include_role:
+        name: network.acls.gather
+      vars:
+        ansible_network_os: cisco.ios.ios
+```
+Example Output
+When the playbook is executed successfully, the output will display the structured facts for the specified resources.
```
Comment (rohitthakur2590) [inline roles/gather/README.md]:
```suggestion
When the playbook is executed successfully, the output will display the structured facts for the ACL resources.
```

File: roles/health_checks/README.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  - `interface_acl_status`: Shows the status of ACL assignments on interfaces.
+
+This role enables users to create a runtime brownfield inventory with all the ACLS configurations in terms of host vars. These host vars are ansible facts that have been gathered through the *acl_interfaces and *acls network resource module. The tasks offered by this role can be observed below:
+
+## Variables
+
+| Variable Name        | Default Value | Required | Type | Description                                                  |       Example       |
+|:---------------------|:-------------:|:--------:|:----:|:-------------------------------------------------------------|:-------------------:|
+| `ansible_network_os` | `""`          | no       | str  | Network OS to be used during detection.                      | `"cisco.ios.ios"` |
+| `data_store`         | `""`          | yes      | dict | Specifies the data store to be used (local or SCM).          | See examples below. |
```
Comment (rohitthakur2590) [inline roles/health_checks/README.md]: I believe we don;t need datastore for health checks

File: tests/integration/targets/acls_remediate/tasks/main.yaml (hunk 1 above)
Comment (AAYUSH2091) [reply tests/integration/targets/acls_remediate/tasks/main.yaml]: Ah yes right The operations variable is no longer needed as we are now focusing on dedicated roles for specific tasks

File: plugins/filter/health_check_view.py [hunk 6]
```diff
@@ -177,18 +177,23 @@ def health_check_view(*args, **kwargs):
...
+    def clean_acl_id(acl_id):
+        """Clean ACL ID by removing backticks and converting to string"""
+        if isinstance(acl_id, (int, str)):
+            return str(acl_id).replace('`', '')
+        return acl_id
+
     if acls_facts.get("interface_data") or acls_facts.get("acls_data"):
         acls = list(acls_facts.get("acls_data", {}).keys())
-
+        print(acls)
```
Comment (rohitthakur2590) [inline plugins/filter/health_check_view.py]: I believe its not required

File: roles/health_checks/README.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+  - `unassigned_acls`: Shows ACLs that are defined but not used on any interface.
+  - `interface_acl_status`: Shows the status of ACL assignments on interfaces.
+
+This role enables users to create a runtime brownfield inventory with all the ACLS configurations in terms of host vars. These host vars are ansible facts that have been gathered through the *acl_interfaces and *acls network resource module. The tasks offered by this role can be observed below:
+
+## Variables
+
+| Variable Name        | Default Value | Required | Type | Description                                                  |       Example       |
+|:---------------------|:-------------:|:--------:|:----:|:-------------------------------------------------------------|:-------------------:|
+| `ansible_network_os` | `""`          | no       | str  | Network OS to be used during detection.                      | `"cisco.ios.ios"`   |
```
Comment (rohitthakur2590) [inline roles/health_checks/README.md]:
```suggestion
| `ansible_network_os` | `""`          | no       | str  | Network OS to be used while performing health-checks.                      | `"cisco.ios.ios"`   |
```

File: roles/health_checks/README.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+**ACL Health Checks** :
+- Enables users to perform health checks for ACL configurations and their interface assignments.
+- Users can perform the following health checks:
+  - `available_acls`: Lists all ACLs configured on the device.
+  - `missing_acls`: Identifies ACLs referenced in interfaces but not defined.
+  - `unassigned_acls`: Shows ACLs that are defined but not used on any interface.
+  - `interface_acl_status`: Shows the status of ACL assignments on interfaces.
+
+This role enables users to create a runtime brownfield inventory with all the ACLS configurations in terms of host vars. These host vars are ansible facts that have been gathered through the *acl_interfaces and *acls network resource module. The tasks offered by this role can be observed below:
```
Comment (rohitthakur2590) [inline roles/health_checks/README.md]: I believe it needs to be corrected w.r.t health-checks

File: roles/health_checks/README.md [hunk 9]
```diff
@@ -0,0 +1,53 @@
+# Health Checks
+The role enables user to manage the ACL resources independent of platforms and perform ACL health checks.
```
Comment (rohitthakur2590) [inline roles/health_checks/README.md]:
```suggestion
The role allows users to perform platform-independent Access Control List (ACL) health checks, ensuring consistent and reliable validation across various networking environments.
```

File: roles/persist/README.md [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+## Features
+- Fetch structured facts for ACL resources and interface assignments.
+- Persist gathered ACL configurations to local directories or remote SCM repositories.
+- Enable centralized and version-controlled storage for network access control policies.
+
+## Variables
+
+| Variable Name        | Default Value | Required | Type | Description                                                   |          Example         |
+|:---------------------|:-------------:|:--------:|:----:|:------------------------------------------------------------- |:------------------------:|
+| `ansible_network_os` | `""`          | no       | str  | Network OS for which the ACL facts are being gathered.        |   `"cisco.ios.ios"`      |
```
Comment (rohitthakur2590) [inline roles/persist/README.md]:
```suggestion
| `ansible_network_os` | `""`          | no       | str  | Network OS for which the ACL facts are being persisted.        |   `"cisco.ios.ios"`      |
```

**Review by rohitthakur2590 — CHANGES_REQUESTED** [review]

File: plugins/filter/health_check_view.py (hunk 6 above)
Comment (AAYUSH2091) [reply plugins/filter/health_check_view.py]: removed print(acls)

File: roles/health_checks/README.md (hunk 8 above)
Comment (AAYUSH2091) [reply roles/health_checks/README.md]: yes this line tells the functionality related to the persist role not the heath_check one.

**Review by rohitthakur2590 — APPROVED** [review]

**Pull request merged**